### PR TITLE
New custom policy DenyAction Delete UAMI used by AMA

### PR DESCRIPTION
**Pull request by arjenhuitema**
## Overview/Summary

New custom policy to safeguard against accidental removal of the User Assigned Managed Identity used by AMA by blocking delete calls using deny action effect.

## This PR fixes/adds/changes/removes

1. Adds Policy Definition DenyAction-DeleteResources
2. Adds Policy Assignment DenyAction-DeleteUAMIAMA
3. Documentation updates 

### Breaking Changes

N/A

## Testing Evidence

Please provide any testing evidence to show that your Pull Request works/fixes as described and planned (include screenshots, if appropriate).

Deployment:
image

Assignment:
image

Action denied:
image

#### Azure Public

[Deploy To Azure](https://portal.azure.com/#view/Microsoft_Azure_CreateUIDef/CustomDeploymentBlade/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FEnterprise-Scale%2Fpatch-policy-denyaction-uami%2FeslzArm%2FeslzArm.json/uiFormDefinitionUri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2FEnterprise-Scale%2Fpatch-policy-denyaction-uami%2FeslzArm%2Feslz-portal.json)

## As part of this Pull Request I have

- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [x] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [x] Performed testing and provided evidence.
- [x] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [x] Updated relevant and associated documentation.
- [x] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

### CODE REVIEW

File: src/resources/Microsoft.Authorization/policyDefinitions/DenyAction-DeleteResources.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
+{
+  "name": "DenyAction-DeleteResources",
+  "type": "Microsoft.Authorization/policyDefinitions",
+  "apiVersion": "2021-06-01",
+  "scope": null,
+  "properties": {
+    "policyType": "Custom",
+    "mode": "All",
+    "displayName": "Do not allow deletion of resource types",
```
Comment (Springstone) [inline src/resources/Microsoft.Authorization/policyDefinitions/DenyAction-DeleteResources.json:9]:
```suggestion
    "displayName": "Do not allow deletion of specified resource and resource type",
```

File: src/resources/Microsoft.Authorization/policyDefinitions/DenyAction-DeleteResources.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    "description": "This policy enables you to specify the resource types that your organization can protect from accidentals deletion by blocking delete calls using deny action effect.",
```
Comment (Springstone) [inline src/resources/Microsoft.Authorization/policyDefinitions/DenyAction-DeleteResources.json:10]:
```suggestion
    "description": "This policy enables you to specify the resource and resource type that your organization can protect from accidentals deletion by blocking delete calls using the deny action effect.",
```

File: docs/wiki/Whats-new.md [hunk 3]
```diff
@@ -45,6 +45,12 @@ This article will be updated as and when changes are made to the above and anyth
 
 Here's what's changed in Enterprise Scale/Azure Landing Zones:
 
+### June 2024
+
+#### Tooling
```
Comment (Springstone) [inline docs/wiki/Whats-new.md:50]:
```suggestion
#### Policy
```

File: docs/wiki/Whats-new.md [hunk 4]
```diff
@@ -45,6 +45,12 @@ This article will be updated as and when changes are made to the above and anyth
...
+
+- Added new custom Policy that provides a safeguard against accidental removal of the User Assigned Managed Identity used by AMA. Assigned at the Platform Management it blocks delete calls using deny action effect.
```
Comment (Springstone) [inline docs/wiki/Whats-new.md:52]:
```suggestion
- Added new custom policy [Do not allow deletion of specified resource and resource type](https://www.azadvertizer.net/azpolicyadvertizer/DenyAction-DeleteResources.html) that provides a safeguard against accidental removal of the User Assigned Managed Identity used by AMA. Assigned at the Platform Management Group, it blocks delete calls using the deny action effect.
```

File: docs/wiki/ALZ-Policies.md [hunk 5]
```diff
@@ -122,7 +122,7 @@ This management group contains all the platform child management groups, like ma
...
-| **Do not allow deletion of resource types**\*                                | **Do not allow deletion of resource types**                              | `Policy Definition`, **Built-in** | This policy enables you to specify the resource types that your organization can protect from accidentals deletion by blocking delete calls using deny action effect. Assigned to deny the deletion of the User Assignment Managed Identity that is used for AMA.                                                              | DenyAction |
+| **Do not allow deletion of the User Assigned Managed Identity used by AMA**\*| **Do not allow deletion of resource types**                              | `Policy Definition`, **Custom** | This policy provides a safeguard against accidental removal of the User Assigned Managed Identity used by AMA by blocking delete calls using deny action effect.                                                                                                                                                                 | DenyAction |
```
Comment (Springstone) [inline docs/wiki/ALZ-Policies.md:125]:
```suggestion
| **Do not allow deletion of the User Assigned Managed Identity used by AMA**\*| **Do not allow deletion of specified resource and resource type**                              | `Policy Definition`, **Custom** | This policy provides a safeguard against accidental removal of the User Assigned Managed Identity used by AMA by blocking delete calls using deny action effect.                                                                                                                                                                 | DenyAction |
```

**Review by Springstone — CHANGES_REQUESTED** [review]
@arjenhuitema looks, great, some minor text adjustments, kindly review.

**Review by Springstone — APPROVED** [review]
LGTM

**Review by jtracey93 — APPROVED** [review]

**Pull request merged**